internal/client.go

```diff
@@ -1003,6 +1031,54 @@ func DialCloudOperationsClient(ctx context.Context, options CloudOperationsClien
 	}, nil
 }
 
+// PrepareUpdateWorkflowOperation returns an UpdateWorkflowOperation that can be used to perform Update-with-Start,
+// or an error in case the update operation is invalid.
+func PrepareUpdateWorkflowOperation(options UpdateWorkflowOptions) (*UpdateWorkflowOperation, error) {
+	input, err := createUpdateWorkflowInput(options)
+	if err != nil {
+		return nil, err
+	}
+	if options.RunID != "" {
+		return nil, errors.New("RunID is not allowed to be used on a prepared update operation")
+	}
+	if options.FirstExecutionRunID != "" {
+		return nil, errors.New("FirstExecutionRunID is not allowed to be used on a prepared update operation")
+	}
+	return &UpdateWorkflowOperation{input: input, doneCh: make(chan struct{})}, nil
+}
+
+// Get blocks until a server response has been received; or the context deadline is exceeded.
+func (op *UpdateWorkflowOperation) Get(ctx context.Context) (WorkflowUpdateHandle, error) {
+	if op.done.Load() {
+		return op.handle, op.err
+	}
+	select {
+	case <-op.doneCh:
+		return op.handle, op.err
+	case <-ctx.Done():
+		return nil, ctx.Err()
+	}
+}
+
+func (op *UpdateWorkflowOperation) execute() error {
+	if op.executing.Swap(true) {
+		return fmt.Errorf("was already executed")
+	}
+	return nil
+}
+
+func (op *UpdateWorkflowOperation) set(handle WorkflowUpdateHandle, err error) {
+	if op.done.Load() {
+		panic("handle has already been set")
+	}
```

Comment:
Not needed in this private method IMO where we know we only set after we mark `execute()`. If you must have this check, consider simply checking whether executing is true by this time (but I don't think it's needed, we can trust ourselves to set it once.